api/server.go

```diff
@@ -674,6 +676,376 @@ func (s *server) walletsFundSFHandler(jc jape.Context) {
 	})
 }
 
+func (s *server) walletsConstructHandler(jc jape.Context) {
+	cs := s.cm.TipState()
+	if cs.Index.Height >= cs.Network.HardforkV2.RequireHeight {
+		jc.Error(errors.New("v1 transactions are not allowed after the v2 require height"), http.StatusBadRequest)
+	}
+
+	var walletID wallet.ID
+	if err := jc.DecodeParam("id", &walletID); err != nil {
+		return
+	}
+	var wcr WalletConstructRequest
+	if err := jc.Decode(&wcr); err != nil {
+		return
+	}
+
+	if wcr.ChangeAddress == types.VoidAddress {
+		jc.Error(errors.New("change address must be specified"), http.StatusBadRequest)
```

Comment:
Any reason why v1 checks for a change address immediately but v2 only does it when there actually is change?